### PR TITLE
Feature/show protected paths

### PR DESCRIPTION
**Pull request by dmitrii-beliakov**
This feature adds calculation of a protected path for each path returned by `/network/paths` API end point and, if the path exists, adds it to the payload under the `protected_path` property.
To enable this functionality a user must specify a request param `include_protected` = true. If this parameter is omitted or false, `network/paths/` behaves as if this feature doesn't exist (nothing new in the response payload).
When a protected path doesn't exist for some path, payload doesn't contain the `protected_path` property (maybe worth to change it and show and empty object to highlight that the logic has been executed).

Closes https://github.com/telstra/open-kilda/issues/5097

### CODE REVIEW

File: src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/PathMapper.java [hunk 1]
```diff
@@ -61,7 +62,14 @@ public org.openkilda.messaging.info.network.Path map(org.openkilda.pce.Path path
             nodes.add(new PathNodePayload(lastSegment.getDestSwitchId(), lastSegment.getDestPort(), null));
         }
 
+        if (path.getProtectedPath() != null && path.getProtectedPath().getProtectedPath() != null) {
```
Comment (pablomuri) [inline src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/PathMapper.java:65]: Why do we have a protected path inside the protected path?

File: src-java/kilda-pce/src/test/java/org/openkilda/pce/mapper/PathSegmentMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.List;
+import java.util.Set;
+
+public class PathSegmentMapperTest {
+
+    @Test
+    public void toPathSegmentTest() {
+        Segment segment1 = Segment.builder()
+                .latency(12L)
+                .srcSwitchId(new SwitchId("00:01"))
```
Comment (pablomuri) [inline src-java/kilda-pce/src/test/java/org/openkilda/pce/mapper/PathSegmentMapperTest.java:39]: [minor] this switchId could be also initialized with an int new SwitchId(1)
```suggestion
                .srcSwitchId(new SwitchId(1)))
```

In any case, I would create the 3 switchIds previously

File: src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/PathMapper.java (hunk 1 above)
Comment (dmitrii-beliakov) [reply src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/mappers/PathMapper.java:65]: a protected path is actually the same type as a path. So a Path type contains a Path type variable; there are checks that don't allow this kind of recursion

File: src-java/kilda-pce/src/main/java/org/openkilda/pce/PathComputer.java [hunk 3]
```diff
@@ -80,6 +80,15 @@ default GetHaPathsResult getHaPath(HaFlow haFlow, boolean isProtected)
     GetHaPathsResult getHaPath(HaFlow haFlow, Collection<PathId> reuseSubPathsResources, boolean isProtected)
             throws UnroutableFlowException, RecoverableException;
 
+    /**
+     * Calculates a protected path based on given parameters and configuration.
```
Comment (IvanChupin) [inline src-java/kilda-pce/src/main/java/org/openkilda/pce/PathComputer.java:84]: should we use 'the' here?
```suggestion
     * Calculates a protected path based on the given parameters and configuration.
```

**Review by IvanChupin — COMMENTED** [review]
Great job! Thanks,
Just a couple of notes from my side.

**Review by IvanChupin — APPROVED** [review]

**Comment by dmitrii-beliakov** [comment]
I fixed the docs and squashed the commits again.

**Pull request merged**